lib/trino-geospatial-toolkit/src/test/java/io/trino/geospatial/rtree/BenchmarkFlatbushQuery.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.geospatial.rtree;
+
+import io.trino.geospatial.Rectangle;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import static io.trino.geospatial.rtree.RtreeTestUtils.makeRectangles;
+
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(2)
+@Warmup(iterations = 3, time = 2, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 10, time = 1, timeUnit = TimeUnit.SECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkFlatbushQuery
+{
+    private static final int SEED = 613;
+    private static final int NUM_PROBE_RECTANGLES = 1000;
+
+    @Benchmark
+    @OperationsPerInvocation(NUM_PROBE_RECTANGLES)
+    public void rtreeQuery(BenchmarkData data, Blackhole blackhole)
+    {
+        for (Rectangle query : data.getProbeRectangles()) {
+            data.getRtree().findIntersections(query, blackhole::consume);
```

Comment:
This occurrence of `Blackhole` is going to be a little trickier to remove since it's semantically important to the measurement and can't otherwise be avoided by, eg: returning a single value. In order to add a test for this, you'll probably have to factor the benchmark method out from some "common" method eg:

```
private void rtreeQueryProbe(BenchmarkData data, ItemVisitor visitor)
{
  // the current method
}

@Benchmark
public void rtreeQuery(BenchmarkData data, Blackhole blackhole)
{
  rtreeQueryProbe(data, blackhole::consume);
}

@Test
public void verifyRtreeQuery()
{
// test rtreeQueryProbe
}
```

Reply:
Done